### ISSUE TITLE
High CPU usage on RaspberryPi using Terminal

### ISSUE THREAD

**Issue by mariodivece** [unosquare/swan#21]
ManualResetEventSlim causes spinwaits (high CPU usage)
Terminal.cs - Lines 21 and 22 need to be changed to:
```csharp
        private static readonly ManualResetEvent OutputDone = new ManualResetEvent(false);
        private static readonly ManualResetEvent InputDone = new ManualResetEvent(true);
```
Change DequeueOutputTask to a background (NOT THREADPOOL PLEASE) Thread with BelowNormalPriority. Here are all the changes:

```csharp
               // Change declaration
               private static readonly Thread DequeueOutputTask;

                // Here we start the output task, fire-and-forget
                DequeueOutputTask = new Thread(DequeueOutputAsync)
                {
                    IsBackground = true,
                    Name = nameof(DequeueOutputTask),
                    Priority = ThreadPriority.BelowNormal
                };

               // Change signature:
               private static void DequeueOutputAsync()

              // wrap the outer loop of DequeueOutputAsync with:
              using (var tickLock = new ManualResetEvent(false))
              {

              // Change await Task.Delay(1) to 
              tickLock.WaitOne(1);
...


I need this fixed to continue one of my projects.
And you need this fixed to continue the AccessCore project as well

**Comment by mariodivece** [comment]
In order to test, run a mono on a Raspberry Pi and compare the CPU usage using TOP via a console application using Console.ReadKey and Terminal.ReadKey

**Comment by geoperez** [comment]
Can you check latest nuget (0.14.10) please?

**Comment by mariodivece** [comment]
It is down to 4.7% ( from 45%). Try a waitone(10) instead of a 1 and let's see if we can make it even more efficient

**Comment by geoperez** [comment]
The change is available at 0.14.11
